### DISCUSSION TITLE
Github Pages not possible to point to /dist folder

### DISCUSSION THREAD

**Discussion by kaspar-allenbach** [community/community#137690] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Folder in Branch

How do I point github pages into that folder?

I have a vue project I want to release via github pages.

the url has to point into the `/dist` folder as far as I understand.
But altough the dist folder is present in the branch nothing shows up

![image](https://github.com/user-attachments/assets/5a95b3c9-1d74-469f-a815-f5b5e8ff3a07)

**Comment by JohnImril** [comment, marked as answer]
To deploy a Vue.js application to GitHub Pages, you can follow these steps:

### 1. **Install the GitHub Pages Deployment Plugin**

You need to install `gh-pages` package to handle the deployment.

```bash
npm install gh-pages --save-dev
```

### 2. **Update `vue.config.js` for GitHub Pages**

Create or update the `vue.config.js` file in your project's root directory with the following content. If your repository is at `https://github.com/<USERNAME>/<REPO>`, you need to specify the correct `publicPath`:

```js
module.exports = {
  publicPath: process.env.NODE_ENV === 'production'
    ? '/<REPO>/'
    : '/'
}
```

Replace `<REPO>` with your GitHub repository name.

### 3. **Update `package.json`**

Add a new script to your `package.json` for deployment:

```json
{
  "scripts": {
    "deploy": "vue-cli-service build && gh-pages -d dist"
  }
}
```

This script will build the project and then deploy the contents of the `dist` folder to GitHub Pages using the `gh-pages` package.

### 4. **Build and Deploy**

Run the following command to build the project and deploy it to GitHub Pages:

```bash
npm run deploy
```

### 5. **Set GitHub Pages Settings**

Go to your GitHub repository:
- Navigate to **Settings** > **Pages**.
- In the **Source** section, choose the `gh-pages` branch as the source.

### 6. **Access Your Deployed Application**

After a few minutes, your Vue.js app should be live at:
```
https://<USERNAME>.github.io/<REPO>
```

**Reply by kaspar-allenbach** [reply]
I use vite. So my correct line would be:

`"deploy": "vite build && gh-pages -d dist"`

**Reply by JohnImril** [reply]
for example i use for vite with react:  
```"scripts": {
		"dev": "vite",
		"build": "tsc -b && vite build",
		"lint": "eslint .",
		"preview": "vite preview",
		"predeploy": "npm run build",
		"deploy": "gh-pages -d dist"
	},```
